internal/collector/sysinfo/hardware/hardware_windows.go

```diff
@@ -169,113 +172,106 @@ func (s Collector) collectMemory() (mem memory, err error) {
 
 // collectDisks uses Win32_DiskDrive and Win32_DiskPartition to collect information about disks.
 func (s Collector) collectDisks() (blks []disk, err error) {
-	var usedDiskFields = map[string]struct{}{
-		"Size":       {},
-		"Index":      {},
-		"Partitions": {},
+	type diskOut struct {
+		MediaType  string
+		Index      uint64
+		Size       uint64
+		Partitions uint64
 	}
 
-	var usedPartitionFields = map[string]struct{}{
-		"DiskIndex": {},
-		"Index":     {},
-		"Size":      {},
+	type partOut struct {
+		DiskIndex uint64
+		Size      uint64
+		Type      string
 	}
 
-	getSize := func(b string) uint64 {
-		v, err := strconv.ParseUint(b, 10, 64)
-		if err != nil {
-			s.log.Warn("disk partition contains invalid size")
-			return 0
-		}
-		v, _ = fileutils.ConvertUnitToStandard("b", v)
-		return v
-	}
-
-	disks, err := cmdutils.RunListFmt(s.platform.diskCmd, usedDiskFields, s.log)
+	stdout, stderr, err := cmdutils.RunWithTimeout(context.Background(), 15*time.Second, s.platform.diskCmd[0], s.platform.diskCmd[1:]...)
 	if err != nil {
+		s.log.Warn("Failed to run disk command", "error", err, "stderr", stderr)
 		return nil, err
 	}
 
-	const maxPartitions = 128
+	if stderr.String() != "" {
+		s.log.Info("disk command returned stderr", "stderr", stderr)
+	}
 
-	blks = make([]disk, len(disks))
-	for _, d := range disks {
-		parts, err := strconv.Atoi(d["Partitions"])
-		if err != nil {
-			s.log.Warn("disk partitions was not an integer", "error", err)
-			parts = 0
-		}
-		if parts < 0 {
-			s.log.Warn("disk partitions was negative", "value", parts)
-			parts = 0
-		}
-		if parts > maxPartitions {
-			s.log.Warn("disk partitions too large", "value", parts)
-			parts = maxPartitions
-		}
+	var disksOut []diskOut
+	if err = json.Unmarshal(stdout.Bytes(), &disksOut); err != nil {
+		s.log.Warn("Failed to unmarshal disk output", "error", err)
+		return nil, err
+	}
 
-		c := disk{
-			Size:     getSize(d["Size"]),
-			Type:     "disk",
-			Children: make([]disk, parts),
+	validDiskIndices := make(map[uint64]bool)
+	diskMap := make(map[uint64]int) // Map between blk index and disk index
+	for _, d := range disksOut {
+		if value, ok := validDiskIndices[d.Index]; ok && value {
+			s.log.Warn("Skipping duplicate disk index", "index", d.Index)
+			continue
 		}
+		validDiskIndices[d.Index] = true
```

Comment:
All unique (or distinct) `d.Index` values are written to this map, there is no other validation involved, thus I'd rename this into `uniqueDiskIndices`, `distinctDiskIndices` or `diskIndicesSeen` because you've already seen them once.